spec/models/new_msa_component_event_spec.rb

```diff
@@ -31,4 +32,24 @@
       expect(event.errors[:environment]).to eql [t('components.errors.invalid_environment')]
     end
   end
-end
+
+  context 'entity_id' do
+    it 'must be provided' do
+      event = build(:new_msa_component_event, entity_id: '')
+      expect(event).to_not be_valid
+      expect(event.errors[:entity_id]).to eql [t('components.errors.missing_entity_id')]
+    end
+
+    it 'is valid when there is only leading and trailing whitespaces' do
+      event = build(:new_msa_component_event, entity_id: ' https://test-entity-id ')
+      expect(event).to be_valid
+      expect(event.errors[:entity_id]).to eql []
+    end
```

Comment:
Can we also check that the whitespaces are stripped? Does the `event` object contain an entity ID string of `'https://test-entity-id'`. This test correctly checks that leading and trailing whitespaces don't fail validation but it doesn't verify that we actually strip them